tests/unit/test_exporter.py

```diff
@@ -23,6 +23,46 @@ def test_parse_config(self, exporter_daemon):
         assert statsd.config["exporter"]["port"].get() == 9748
         assert statsd.config["exporter"]["collect_interval"].get() == 15
 
+    @pytest.mark.asyncio
+    async def test_update_registry(self, monkeypatch, exporter_daemon):
+        """Test update_registry function."""
+        statsd = exporter_daemon()
+        collector = statsd.collector
+        stats = await collector.get_stats()
+        sample_labels = [
+            {
+                "job": "prometheus-juju-exporter",
+                "hostname": "hostname0",
+                "customer": "customer0",
+                "cloud_name": "cloud name",
+                "juju_model": "juju model",
+                "type": "machine type",
+            },
+            {
+                "job": "prometheus-juju-exporter",
+                "hostname": "hostname1",
+                "customer": "customer1",
+                "cloud_name": "cloud name",
+                "juju_model": "juju model",
+                "type": "machine type",
+            },
+        ]
+        metric1, metric2 = mock.MagicMock(), mock.MagicMock()
+        metrics = [metric1, metric2]
+        sample1, sample2 = mock.MagicMock(), mock.MagicMock()
+        sample1.labels, sample2.labels = sample_labels
+        samples = [sample1, sample2]
+        for sample in samples:
+            sample.name = "example_gauge"
+        for metric in metrics:
+            metric.samples = samples
+        monkeypatch.setattr(
+            "prometheus_juju_exporter.exporter.CollectorRegistry.collect",
+            lambda r: metrics,
+        )
+        statsd.update_registry(stats)
+        assert len(statsd.metrics["example_gauge"].call_labels) == 2
```

Comment:
not-blocking: maybe we could be more specific here

Reply:
I've updated it